File: packages/domain/src/health-journal.ts
export interface HealthJournalEntry {
  id: string;
  entry_date: string;
  energy_level: number | null;
  mood_level: number | null;
  sleep_quality: number | null;
  stress_level: number | null;
  symptoms: string[] | null;
  notes: string | null;
  created_at: string;
  updated_at: string;
}

export interface HealthJournalEntryCreate {
  entry_date: string;
  energy_level?: number | null;
  mood_level?: number | null;
  sleep_quality?: number | null;
  stress_level?: number | null;
  symptoms?: string[] | null;
  notes?: string | null;
}

export interface HealthJournalEntryUpdate {
  energy_level?: number | null;
  mood_level?: number | null;
  sleep_quality?: number | null;
  stress_level?: number | null;
  symptoms?: string[] | null;
  notes?: string | null;
}

export interface HealthJournalTrendPoint {
  date: string;
  value: number;
}

export interface HealthJournalSummary {
  start_date: string;
  end_date: string;
  entry_count: number;
  avg_energy: number | null;
  avg_mood: number | null;
  avg_sleep: number | null;
  avg_stress: number | null;
  symptom_frequency: Record<string, number>;
  trend_energy: HealthJournalTrendPoint[];
  trend_mood: HealthJournalTrendPoint[];
  trend_sleep: HealthJournalTrendPoint[];
}

File: apps/api/app/schemas/health_journal.py
"""Pydantic schemas for health journal entries."""

from datetime import date, datetime
from uuid import UUID

from pydantic import BaseModel, Field


class HealthJournalEntryCreate(BaseModel):
    entry_date: date
    energy_level: int | None = Field(None, ge=1, le=10)
    mood_level: int | None = Field(None, ge=1, le=10)
    sleep_quality: int | None = Field(None, ge=1, le=10)
    stress_level: int | None = Field(None, ge=1, le=10)
    symptoms: list[str] | None = None
    notes: str | None = None


class HealthJournalEntryUpdate(BaseModel):
    energy_level: int | None = Field(None, ge=1, le=10)
    mood_level: int | None = Field(None, ge=1, le=10)
    sleep_quality: int | None = Field(None, ge=1, le=10)
    stress_level: int | None = Field(None, ge=1, le=10)
    symptoms: list[str] | None = None
    notes: str | None = None


class HealthJournalEntryResponse(BaseModel):
    model_config = {"from_attributes": True}

    id: UUID
    entry_date: date
    energy_level: int | None = None
    mood_level: int | None = None
    sleep_quality: int | None = None
    stress_level: int | None = None
    symptoms: list[str] | None = None
    notes: str | None = None
    created_at: datetime
    updated_at: datetime


class HealthJournalSummaryResponse(BaseModel):
    start_date: date
    end_date: date
    entry_count: int
    avg_energy: float | None = None
    avg_mood: float | None = None
    avg_sleep: float | None = None
    avg_stress: float | None = None
    symptom_frequency: dict[str, int] = {}
    trend_energy: list[dict] = []
    trend_mood: list[dict] = []
    trend_sleep: list[dict] = []
